src/Parsers/Kusto/ParserKQLStatement.cpp

```diff
@@ -95,8 +97,59 @@ bool ParserKQLTaleFunction::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
             return true;
         }
     }
-    pos =  begin;
+    pos = begin;
     return false;
-};
+};*/
+
+bool ParserKQLTableFunction::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserKQLWithUnionQuery kql_p;
+    ASTPtr select;
+    ParserToken s_lparen(TokenType::OpeningRoundBracket);
+
+    auto begin = pos;
+    auto paren_count = 0;
+    String kql_statement;
+
+    if (s_lparen.ignore(pos, expected))
+    {
+        if (pos->type == TokenType::HereDoc)
+        {
+            auto kql_table_str = String(pos->begin, pos->end);
+            size_t pos1 = kql_table_str.find("$$");
+            size_t pos2 = kql_table_str.rfind("$$");
+            if (pos1 != std::string::npos && pos2 != std::string::npos && pos1 < pos2)
+                kql_statement = kql_table_str.substr(pos1+2, pos2 - pos1 -2);
+        }
+        else
+        {
+            ++paren_count;
+            auto pos_start = pos;
+            while (!pos->isEnd())
+            {
+                if (pos->type == TokenType::ClosingRoundBracket)
+                    --paren_count;
+                if (pos->type == TokenType::OpeningRoundBracket)
+                    ++paren_count;
+
+                if (paren_count == 0)
+                    break;
+                ++pos;
+            }
+            kql_statement = String(pos_start->begin, (--pos)->end);
+        }
```

Comment:
This will allow `kql(<naked kql expression>)` - which means kql expression will be processed by tokenizer - currently it seems to work fine, but in the future, if we will change something in tokenizer or you will introduce some incompatible syntax for kql, it can break. You can leave it this way of course since it will only impact `kql()` statement, but I strongly recommend to remove this option and left only `kql($$<kql expression>$$)` variant.

Reply:
yes, this part will be removed, we leave it here just  for our client to have a buffer time make changes, as long as they complete their  changes, this part will be removed